server/src/main/java/org/apache/cassandra/sidecar/db/SystemAuthDatabaseAccessor.java

```diff
@@ -146,11 +156,26 @@ public Map<String, Boolean> findAllRolesToSuperuserStatus()
     {
         BoundStatement statement = tableSchema.allRoles().bind();
         ResultSet result = execute(statement);
-        Map<String, Boolean> roles = new HashMap<>();
-        for (Row row : result)
+        List<Row> rows = result.all();
+        Map<String, Boolean> roleToSuperUser = rows.stream()
```

Comment:
Would it make sense to actually have a cached version of this?

Reply:
we cache it temporarily. For authorization we cache the super user status